receiver/prometheusreceiver/internal/metrics_adjuster_test.go

```diff
@@ -317,6 +317,8 @@ func Test_jobGC(t *testing.T) {
 	time.Sleep(2 * gcInterval)
 	// re-run job 2, round1 - trigger job gc, removing unmarked entries
 	runScript(t, jobsMap.get("job", "1"), job2Script1)
+	// sleep longer than gcInterval to enable job gc in the next run
```

Comment:
@dinooliva my understanding of the test was that it was missing this sleep since after my fix to the race it is inconsistent about passing this.

Reply:
Side note: we could add a component to wrap the real clock so we could make time move forward without actual sleeps, but, that is a separate issue.

Reply:
sleep is not necessary because for job "0" already 4*gcIntereval has elapsed.
What is the failure? Is it not removing data from job1Script1

Reply:
If I remove the sleep I get it to consistently fail with the following command:

Reply:
After re-running job 2, script 1, gc() will remove the entries. On the mac, it looks like the gc() go routine runs before executing job 1, script 2 - if the gc() go routine doesn't run first, the test will fail. Sleeping for an instance should fix the issue and it's likely fine but w/o adding a signal (or similarly) to gc() we can't guarantee that. One alternative would be to call jobsMap.gc() directly.

Reply:
Experimented with `jobsMap.gc()` replacing the sleep. It passed my local tests. Should I replace the other sleeps too?

Reply:
The other sleeps are necessary for ageing-out the map entries - this is the only one you need to make into an explicit call to gc().